packages/services/rate-limit/src/api.ts

```diff
@@ -1,50 +1,149 @@
-import { z } from 'zod';
-import { handleTRPCError } from '@hive/service-common';
-import type { FastifyRequest } from '@hive/service-common';
-import type { inferRouterInputs, inferRouterOutputs } from '@trpc/server';
-import { initTRPC } from '@trpc/server';
+import {
+  createRouter,
+  FromSchema,
+  Response,
+  RouterInput,
+  RouterOutput,
+  useErrorHandling,
+} from 'fets';
+import { createLogger } from '@graphql-yoga/logger';
+import { reportReadiness } from '@hive/service-common';
+import * as Sentry from '@sentry/node';
 import type { Limiter } from './limiter';
 
 export interface Context {
-  req: FastifyRequest;
   limiter: Limiter;
 }
 
-const t = initTRPC.context<Context>().create();
-const errorMiddleware = t.middleware(handleTRPCError);
-const procedure = t.procedure.use(errorMiddleware);
+const VALIDATION = {
+  type: 'object',
+  properties: {
+    id: {
+      type: 'string',
+      minLength: 1,
+    },
+    entityType: {
+      type: 'string',
+      enum: ['organization', 'target'],
+    },
+    type: {
+      type: 'string',
+      enum: ['operations-reporting'],
+    },
+    token: {
+      type: 'string',
+      nullable: true,
+    },
+  },
+  additionalProperties: false,
+  required: ['id', 'entityType', 'type', 'token'],
+} as const;
 
-export type RateLimitInput = z.infer<typeof VALIDATION>;
+export type RateLimitInput = FromSchema<typeof VALIDATION>;
 
-const VALIDATION = z
-  .object({
-    id: z.string().min(1),
-    entityType: z.enum(['organization', 'target']),
-    type: z.enum(['operations-reporting']),
-    /**
-     * Token is optional, and used only when an additional blocking (WAF) process is needed.
-     */
-    token: z.string().nullish().optional(),
-  })
-  .required();
-
-export const rateLimitApiRouter = t.router({
-  getRetention: procedure
-    .input(
-      z
-        .object({
-          targetId: z.string().nonempty(),
-        })
-        .required(),
-    )
-    .query(({ ctx, input }) => {
-      return ctx.limiter.getRetention(input.targetId);
-    }),
-  checkRateLimit: procedure.input(VALIDATION).query(({ ctx, input }) => {
-    return ctx.limiter.checkLimit(input);
-  }),
-});
+export const logger = createLogger();
 
 export type RateLimitApi = typeof rateLimitApiRouter;
-export type RateLimitApiInput = inferRouterInputs<RateLimitApi>;
-export type RateLimitApiOutput = inferRouterOutputs<RateLimitApi>;
+export type RateLimitApiInput = RouterInput<RateLimitApi>;
+export type RateLimitApiOutput = RouterOutput<RateLimitApi>;
+
+export const rateLimitApiRouter = createRouter<Context>({
+  title: 'Rate Limit API',
+  version: '0.0.0',
```

Comment:
can we avoid the need for these by default? (title, version)

Reply:
Yes of course, I just want to show them :)